sys/cddl/dev/dtrace/dtrace_ioctl.c

```diff
@@ -214,32 +248,51 @@ dtrace_ioctl(struct cdev *dev, u_long cmd, caddr_t addr,
 		return (0);
 	}
 	case DTRACEIOC_AGGSNAP:
-	case DTRACEIOC_BUFSNAP: {
-		dtrace_bufdesc_t **pdesc = (dtrace_bufdesc_t **) addr;
-		dtrace_bufdesc_t desc;
+	case DTRACEIOC_BUFSNAP:
+	case DTRACEIOC_BUFSNAP_64: {
+		dtrace_bufdesc_t *__capability *pdesc =
```

Comment:
`pdesc` goes unused after assignment to `pdesc_cap`. Perhaps coalesce with the `pdesc_cap` assignment or in a separate copyin function?